src/Microsoft.Identity.Client/TokenCache.cs

```diff
@@ -1043,35 +1053,41 @@ void ITokenCacheInternal.RemoveMsalAccount(IAccount account, RequestContext requ
                 return;
             }
 
-            // Delete ALL refresh tokens associated with this account
             var allRefreshTokens = ((ITokenCacheInternal)this).GetAllRefreshTokens(false)
                 .Where(item => item.HomeAccountId.Equals(account.HomeAccountId.Identifier, StringComparison.OrdinalIgnoreCase))
                 .ToList();
 
-            foreach (MsalRefreshTokenCacheItem refreshTokenCacheItem in allRefreshTokens)
+            // To maintain backward compatiblity with other MSALs, filter all credentials by clientID if
+            // Foci is disabled or if an FRT is not present
+            bool filterByClientId = !_featureFlags.IsFociEnabled || !FrtExists(allRefreshTokens);
```

Comment:
@oldalton @henrik-me  - I made another change that can be summarized as: "If an FRT is not present, filter by client If an FRT is present, do not filter by clientID. ". This means that MSAL.NET is now consistent with all other MSALs.

Functionally, these scenarios are mostly theoretical, but unit tests do cover them. We could look at strengthening these conditions some more, but now that MSAL.net is consistent with other MSALs, any improvements and bugs will need to be prioritized with other bugs.